packages/unmock-core/src/service/dsl/dsl.ts

```diff
@@ -1,42 +1,48 @@
 import debug from "debug";
 import { cloneDeep } from "lodash";
-import {
-  codeToMedia,
-  isReference,
-  mediaTypeToSchema,
-  Schema,
-} from "../interfaces";
+import { codeToMedia, Schema } from "../interfaces";
+import { actOn$times } from "./actors";
+import { SCHEMA_TIMES } from "./constants";
 import { ITopLevelDSL } from "./interfaces";
+import { translate$size, translate$times } from "./translators";
+import {
+  hasUnmockProperty,
+  injectUnmockProperty,
+  throwOnErrorIfStrict,
+} from "./utils";
 
 const debugLog = debug("unmock:dsl");
 
-type Props = Record<string, Schema>;
-const SCHEMA_TIMES = "x-unmock-times";
-const UNMOCK_TYPE = "unmock";
-const buildUnmockProperty = (type: string, value: any) => ({
-  [type]: { type: UNMOCK_TYPE, default: value },
-});
-const hasUnmockProperty = (schema: Schema, type: string) => {
-  const log = (found: boolean) =>
-    debugLog(
-      `${found ? "Found" : "Can't find"} '${type}' in ${JSON.stringify(
-        schema,
-      )}`,
-    );
-  if (schema.properties === undefined) {
-    log(false);
-    return false;
-  }
-  const prop = schema.properties[type];
-  if (prop === undefined || isReference(prop)) {
-    log(false);
-    return false;
-  }
-  log(prop.type === UNMOCK_TYPE);
-  return prop.type === UNMOCK_TYPE;
-};
+/**
+ * Handles DSL arguments by translating them to OAS where needed.
+ * Many of these static functions modify their inputs to abstract away
+ * the DSL from the final output (after having it translated/acted upon).
+ */
+export abstract class DSL {
+  /**
+   * If true, mismatching arguments will throw
+   *    (e.g. `$size: a`, `$size: 0.1` (rounds to 0), `$size: N` with non-array type)
+   * If false, these arguments are considered "as if" when they don't match the schema.
+   */
+  public static STRICT_MODE = true;
```

Comment:
Not sure if we want to expose this to end users. I put it here for the time being so it's easy to switch between the modes internally and decide what fits best for the flow.
We could expose it a-la `unmock.strict_dsl()` or `unmock.lax_dsl()`?